src/App.tsx

```diff
@@ -1,24 +1,256 @@
+        setCorrectAnswer(questions[number].correct_answer);
+
+        // to give the  wrong answer a red background
+        setWrongAnswer(answer)
+      }
+
+
+      // save answer in answerObject
+      // save answer in object
+      const AnswerObject = {
+        question : questions[number].question,
+        answer,
+        correct,
+        correctAnswer : questions[number].correct_answer
+      }
+      // add answer above to userAnswer
+      setUserAnswers((prev) => [...prev, AnswerObject]);
+
+    }
+  };
+
+  const nextQuestion = () => {
+
+    if(number + 1 === parseInt(totalQuestions, 10)){
+      setGameOver(true);
+    }
+    else {
+      setNumber(prev => prev + 1); 
+    }
+  }
+
+  const settingsForm = (e: React.MouseEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    startTrivia();
+  }
+  
+  const handleDifficultyChange = (e: any) => {
+    setDifficulty(e.currentTarget.value)
+  }
+
+  const handleCategoryChange = (e: any) => {
+    setCategory(e.currentTarget.value)
+  }
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="quiz-app">
+      <h1> QUIZ APP </h1>
+
+      {
+        gameOver || (number + 1 === parseInt(totalQuestions, 10)) ?
+        <div className="start-settings">
+          {
+            !settings &&
+            <button className="start" onClick={startTrivia}> start </button>
+          }
+          {
+            !settings &&
+            <button className="start" onClick={trueSettings} disabled={settings}> settings </button>
+          }
+          </div>
+          : null
+      }
+
+      {
+        !gameOver && !settings &&
+          <p className="score"> score: {score}</p>
+      }
+
+      {
+        loading &&
+          <p> Loading Questions ...</p>
+      }
+
+      {
+        !loading && !gameOver && !settings &&
+          <QuestionCard
+            questionNr={number + 1}
+            totalQuestions={totalQuestions}
+            question={questions[number].question}
+            answers={questions[number].answer}
+            userAnswer={userAnswers ? userAnswers[number] : undefined}
+            callback={checkAnswer}
+            correctAnswer={correctAnswer}
+            wrongAnswer={wrongAnswer}
+          />
+      }
+
+      {
+        settings && 
+        <div className="settings-block">
+          <button className="start" disabled={settings}> settings </button>
+          <form onSubmit={settingsForm} className="form">
+            <div className="settings-block">
+              <label>Number of Questions</label>
+              <input 
```

Comment:
I think it's better to group these as:
```
              <input
                required
                type="number"
                max="50"
                placeholder="maximum number is 50"
                value={totalQuestions}
                id="totalQuestions"
                className="totalquestion"
                onChange={(e:React.ChangeEvent<HTMLInputElement>) => 
                  setTotalQuestions(e.currentTarget.value)}
                />
```

This is because the props above are in a more logical sequence than before i.e "give me a required number input that maxes out at 50, has the following values, has an id and className of so and so, and has the following onChange handler.".